specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2021-01-01/ResourceHealth.json

```diff
@@ -1164,6 +1236,402 @@
+        },
+        "state": {
+          "type": "string",
+          "description": "Incident state."
+        },
+        "createdDate": {
+          "type": "string",
+          "description": "Incident created date."
+        },
+        "sourceCreateTime": {
+          "type": "string",
+          "description": "The date on which source created incident."
+        },
+        "lastModifiedDate": {
+          "type": "string",
+          "description": "Incident last modified date."
+        },
+        "title": {
+          "type": "string",
+          "description": "Title for the incident."
+        },
+        "summary": {
+          "type": "string",
+          "description": "Summary for the incident."
+        },
+        "description": {
+          "type": "string",
+          "description": "Description for the incident."
+        },
+        "cloudInstanceId": {
+          "type": "integer",
+          "description": "CloudInstanceId or TenantSiloId for the incident 's tenant.",
+          "format": "int64"
+        },
+        "responsibleServiceId": {
+          "type": "integer",
+          "description": "Service responsible for the incident.",
+          "format": "int64"
+        },
+        "responsibleTeamId": {
+          "type": "integer",
+          "description": "Team responsible for the incident.",
+          "format": "int64"
+        },
+        "owningServiceId": {
+          "type": "integer",
+          "description": "Owning service identifier for the incident.",
+          "format": "int64"
+        },
+        "serviceCategoryId": {
+          "type": "integer",
+          "description": "Id of the service category of the service that owns this incident.",
+          "format": "int64"
+        },
+        "owningTeamId": {
+          "type": "integer",
+          "description": "Owning team identifier for the incident.",
+          "format": "int64"
+        },
+        "originatingServiceId": {
+          "type": "integer",
+          "description": "Originating service identifier for the incident.",
+          "format": "int64"
+        },
+        "isCustomerImpacting": {
+          "type": "boolean",
+          "description": "A value indicating whether this instance is customer impacting."
+        },
+        "isNoise": {
+          "type": "boolean",
+          "description": "A value indicating whether this instance is noise."
+        },
+        "isSecurityRisk": {
+          "type": "boolean",
+          "description": "A value indicating whether this instance is security risk."
+        },
+        "keywords": {
+          "type": "string",
+          "description": "Keywords for the incident."
+        },
+        "type": {
+          "type": "string",
+          "description": "Incident type."
+        },
+        "parentId": {
+          "type": "integer",
+          "description": "Parent incident id.",
+          "format": "int64"
+        },
+        "impactedServices": {
```

Comment:
>impactedServices [](start = 9, length = 16)

I noticed that your API has quite a few opaque identifiers. How would a user know that 239034 here means "Service Foo" in ICM? (And in general, what is the motivation for surfacing this sort of data in ARM.)

Reply:
this api is meant to surface POTENTIAL outages data along with their caused ICM tickets. in the first phase, this data will be primarily used by support engineers to decrease/eliminate engineering time on certain support cases. thus, both data, e.g. outages and icm incidents will be important. in the future, this data can used on portal to allow customers to have more visibility on their own issues. let me know if that answers your question.

Reply:
Wouldn't the opaque identifiers be a problem? The API won't be fully usable without access to the ICM API, no? In whatever future portal experience, displaying an integer for an impacted service or parent incident won't really be helpful without providing a display name for each one. Such info isn't available in the API you're proposing here, so the portal would have to interact with the ICM APIs as well. (And how would auth even work in these cases? ICM does not use AAD.)

My main concern here is that you're setting yourselves up for additional challenges and likely breaking changes to this API in the future. Thoughts?

Reply:
I've got some new development regarding the data properties. I had a misunderstanding that all the data returned from BRAIN need to be surfaced, which isn't the case here. I will push those changes as soon as it's ready.

Reply:
the new commit should address the concerns.

Reply:
Makes sense.

---
In reply to: [554133726](https://github.com/Azure/azure-rest-api-specs/pull/12206#discussion_r554133726) [](ancestors = 554133726)